components/provisioner/internal/gardener/shoot_reconciler.go

```diff
@@ -264,3 +284,62 @@ func (r *ProvisioningOperator) setDeprovisioningFinished(cluster model.Cluster,
 
 	return nil
 }
+
+func setAuditConfig(shoot *gardener_types.Shoot, policyConfigMapName, subAccountId string) {
+	if shoot.Spec.Kubernetes.KubeAPIServer == nil {
+		shoot.Spec.Kubernetes.KubeAPIServer = &gardener_types.KubeAPIServerConfig{}
+	}
+
+	shoot.Spec.Kubernetes.KubeAPIServer.AuditConfig = &gardener_types.AuditConfig{
+		AuditPolicy: &gardener_types.AuditPolicy{
+			ConfigMapRef: &v12.ObjectReference{Name: policyConfigMapName},
+		},
+	}
+
+	annotate(shoot, auditLogsAnnotation, subAccountId)
+}
+
+func (r *Reconciler) enableAuditLogs(shoot *gardener_types.Shoot, policyConfigMapName, seed string) error {
+	logrus.Info("Enabling audit logs")
+	tenant, err := r.getAuditLogTenant(seed)
+	if err != nil {
+		return err
+	}
+
+	if tenant != "" {
```

Comment:
I think we could also check the current audit log tenant on Shoot and update only if they are not equal.
We can also flip the if statement to make it a little more clear.
```go
if tenant == "" {
    logrus.Warnf("Cannot enable audit logs. Tenant for region %s is empty", seed)
    return nil
} else if tenant == currentTenant {
    logrus.Debugf("Seed %s did not changed, skipping annotating with Audit Log Tenant", seed)
    return nil
}

logrus.Infof("Modifying Audit Log Tenant") // We can log old one and new one but I am not sure how confidential are those

return r.provisioningOperator.updateShoot(*shoot, func(s *gardener_types.Shoot) {
			setAuditConfig(s, policyConfigMapName, tenant)
		})
```

Reply:
I would skip logging it, added info for which shoot the tenant was changed (this is absolutely sufficient).